website/docs/configuration/version-constraints.html.md

```diff
@@ -0,0 +1,95 @@
+---
+layout: "docs"
+page_title: "Version Constraints - Configuration Language"
+---
+
+# Version Constraints
+
+Anywhere that Terraform lets you specify a range of acceptable versions for
+something, it expects a specially formatted string known as a version
+constraint. Version constraints are used when configuring:
+
+- [Modules](./modules.html)
+- [Provider requirements](./provider-requirements.html)
```

Comment:
Should we include provider configuration blocks here as places where version constraints can be used? 

Since we recommend against it I'm perfectly happy omitting it, especially if there's already a link from the provider configuration reference to this page, but I wanted to ask for the sake of completeness.

Reply:
This content was written by Nick rather than me and I didn't significantly change it, but I do like excluding that case here given that it's been non-recommended for an entire major release now, and we still have [the 0.11-and-earlier docs](https://www.terraform.io/docs/configuration-0-11/providers.html#provider-versions) up on the site for folks using versions where `required_providers` didn't exist yet, including 0.11-flavored version constraint documentation.

The docs about the `provider` block do mention in passing that there _used to be_ a `version` argument and say not to use it, which feels like the right level of coverage of that feature to me now that we're a whole major version away from our original recommendation not to use it: I left it in only as a prompt for someone who is looking at an old Terraform module that used `version` inside a `provider` block and wants to find the documentation for what that used to mean.